test/sync/src/commonMain/kotlin/io/realm/test/mongodb/util/AdminApi.kt

```diff
@@ -226,6 +261,98 @@ open class AdminApiImpl internal constructor(
         }
     }
 
+    override suspend fun setAutomaticConfirmation(enabled: Boolean) {
+        val providerId: String = getLocalUserPassProviderId()
+        val url = "$url/groups/$groupId/apps/$appId/auth_providers/$providerId"
+
+        // Fetch current config and update "autoConfirm" property
+        var authProviderConfig: JsonObject = client.typedRequest<JsonObject>(Get, url)
```

Comment:
Depending on the number of fields, etc. you could consider to define a class for the configuration to handle automatic serialization of these request/response types as done for `LoginResponse`, `Role`, etc. 

There was some issue (ktor/kotlinx-serialization/native/JsonArray-type...) that forced me into just using raw JsonObjects, but it was in a huge pool of other stuff that didn't play well, so maybe not worth a re-dive (🕳️ 🐇)

Reply:
I did consider it, but it seems I then have to model all JSON objects in a type-safe way using classes, which also seems like a pretty big overkill. Even though the Kotlin JSON API isn't the best, just manipulating the Maps directly seemed the easiest. Especially as we don't expect them to change.